hat/assets/js/apps/Iaso/domains/forms/fields/hooks/useGetQueryBuildersFields.ts

```diff
@@ -143,29 +143,44 @@ export const useGetQueryBuildersFields = (
     );
 };
 
+export const useGetQueryBuildersFields = (
+    formDescriptors?: FormDescriptor[],
+    possibleFields?: PossibleField[],
+    configFields: Field[] = iasoFields,
+): QueryBuilderFields => {
+    const { formatMessage } = useSafeIntl();
+    return getQueryBuildersFields(
+        formatMessage,
+        formDescriptors,
+        possibleFields,
+        configFields,
+    );
+};
+
 export const useGetQueryBuilderFieldsForAllForms = (
     formDescriptors?: FormDescriptor[],
     allPossibleFields?: PossibleFieldsForForm[],
 ): QueryBuilderFields => {
+    const { formatMessage } = useSafeIntl();
     if (!allPossibleFields || !formDescriptors) return {};
-    const fields: QueryBuilderFields = {};
-
-    for (const { form_id, name, possibleFields } of allPossibleFields) {
-        const subfields = useGetQueryBuildersFields(
-            formDescriptors,
-            possibleFields,
-        );
-
-        fields[form_id] = {
-            label: name,
-            type: '!group',
-            mode: 'array',
-            conjunctions: ['AND', 'OR'],
-            operators: ['some', 'all', 'none'],
-            defaultOperator: 'some',
-            subfields,
-        };
-    }
-
-    return fields;
+    return allPossibleFields.reduce(
+        (fields, { form_id, name, possibleFields }) => ({
+            ...fields,
+            [form_id]: {
+                label: name,
+                type: '!group',
+                mode: 'array',
+                conjunctions: ['AND', 'OR'],
+                operators: ['some', 'all', 'none'],
+                defaultOperator: 'some',
+                subfields: getQueryBuildersFields(
+                    formatMessage,
+                    formDescriptors,
+                    possibleFields,
+                    iasoFields,
+                ),
+            },
+        }),
+        {} as QueryBuilderFields,
+    );
```

Comment:
This leverages TS better

```suggestion
    const reducedFields : QueryBuilderFields = {}
    return allPossibleFields.reduce(
        (fields, { form_id, name, possibleFields }) => ({
            ...fields,
            [form_id]: {
                label: name,
                type: '!group',
                mode: 'array',
                conjunctions: ['AND', 'OR'],
                operators: ['some', 'all', 'none'],
                defaultOperator: 'some',
                subfields: getQueryBuildersFields(
                    formatMessage,
                    formDescriptors,
                    possibleFields,
                    iasoFields,
                ),
            },
        }),
        reducedFields,
    );
```